File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/punishment/PunishmentPostDTO.java
package es.jocasolo.competitiveeventsapi.dto.punishment;

import java.io.Serializable;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;

import es.jocasolo.competitiveeventsapi.dto.DTO;
import es.jocasolo.competitiveeventsapi.enums.score.ScoreSortType;
import lombok.Getter;
import lombok.Setter;

public class PunishmentPostDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	@Getter
	@Setter
	@NotBlank
	private String title;
	
	@Getter
	@Setter
	private String description;
	
	@Getter
	@Setter
	@NotBlank
	private String eventId;
	
	@Getter
	@Setter
	@NotNull
	private Integer requiredPosition;
	
	@Getter
	@Setter
	@NotNull
	private ScoreSortType sortScore;

	@Override
	public String toString() {
		return "RewardPostDTO [title=" + title + "]";
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/punishment/PunishmentPostDTO.kt
package es.jocasolo.competitiveeventsapp.dto.punishment

import com.google.gson.annotations.SerializedName
import es.jocasolo.competitiveeventsapp.dto.BackStackEntryDTO
import es.jocasolo.competitiveeventsapp.enums.score.ScoreSortType

class PunishmentPostDTO(
        @SerializedName("title") var title: String,
        @SerializedName("description") var description: String? = null,
        @SerializedName("eventId") var eventId: String? = null,
        @SerializedName("sortScore") var sortScore: ScoreSortType? = null,
        @SerializedName("requiredPosition") var requiredPosition: Int? = null,
        @SerializedName("id") var id: Int? = null,
) : BackStackEntryDTO() {
}